frontend/src/slices/annotator/annotations.ts

```diff
@@ -0,0 +1,295 @@
+    validation: null
+  }
+}
+
+
+export const annotationsSlice = createSlice({
+  name: 'Annotations',
+  initialState: {
+    currentMode: AnnotationMode.wholeFile,
+    hasChanged: false,
+    wholeFileBoundaries: {
+      startTime: '',
+      startFrequency: 0,
+      endTime: '',
+      endFrequency: 0,
+      duration: 0
+    },
+    results: [],
+    taskComment: DEFAULT_COMMENT,
+    allTags: [],
+    presenceTags: [],
+    allConfidences: [],
+    tagColors: {},
+  } as AnnotationsSlice,
+  reducers: {
+    initAnnotations: (state, action: { payload: Retrieve }) => {
+      const taskComment = action.payload.taskComment.length > 0 ? action.payload.taskComment[0] : {
+        comment: '',
+        annotation_task: action.payload.id,
+        annotation_result: null
+      }
+      const tagColors = {};
+      for (const tag of action.payload.annotationTags) {
+        Object.assign(tagColors, {[tag]: COLORS[action.payload.annotationTags.indexOf(tag) % COLORS.length]})
+      }
+      Object.assign(state, {
+        hasChanged: false,
+        currentMode: action.payload.annotationScope,
+        results: action.payload.prevAnnotations.map(a => {
+          const isBox = isBoxResult(a);
+          return {
+            type: isBox ? AnnotationType.box : AnnotationType.tag,
+            id: a.id,
+            annotation: a.annotation,
+            confidenceIndicator: a.confidenceIndicator,
+            startTime: isBox ? a.startTime ?? 0 : -1,
+            endTime: isBox ? a.endTime ?? 0 : -1,
+            startFrequency: isBox ? a.startFrequency ?? 0 : -1,
+            endFrequency: isBox ? a.endFrequency ?? 0 : -1,
+            result_comments: a.result_comments,
+            validation: action.payload.mode === 'Create' ? null : (a.validation === null ? true : !!a.validation)
+          }
+        }),
+        focusedResult: undefined,
+        focusedComment: taskComment,
+        taskComment,
+        allTags: action.payload.annotationTags,
+        presenceTags: action.payload.prevAnnotations.map(a => a.annotation),
+        tagColors,
+        focusedTag: undefined,
+        allConfidences: action.payload.confidenceIndicatorSet?.confidence_indicators.map(c => c.label) ?? [],
+        confidenceDescription: action.payload.confidenceIndicatorSet?.desc,
+        focusedConfidence: action.payload.confidenceIndicatorSet?.confidence_indicators.find(c => c.isDefault)?.label
+          ?? action.payload.confidenceIndicatorSet?.confidence_indicators.find(c => c.label)?.label,
+        wholeFileBoundaries: action.payload.boundaries
+      });
+    },
+
+    focusResult: (state, action: { payload: Annotation }) => {
+      Object.assign(state, {
+        focusedResult: action.payload,
+        focusedComment: action.payload.result_comments.length > 0 ? action.payload.result_comments[0] : undefined,
+        focusedTag: action.payload.annotation,
+        focusedConfidence: action.payload.confidenceIndicator
+      });
+    },
```

Comment:
Just a comment, nothing to change here: I'll have to get used trhe redux/toolkit way of doing things. I had to check the documentation because I was afraid of this code "mutating" the state, but the documentation knew I would react like I did:

> This object will be passed to [createReducer](https://redux-toolkit.js.org/api/createReducer), so the reducers may safely "mutate" the state they are given.